.gitignore

```diff
@@ -44,7 +44,11 @@ app.*.map.json
 /android/app/debug
 /android/app/profile
 /android/app/release
+*/pubspec.lock
+
 packages/presentation/pubspec.lock
 packages/domain/pubspec.lock
-pubspec.lock
+packages/data/pubspec.lock
+.vscode/launch.json
 
+pubspec.lock
```

Comment:
A leading ** followed by a slash means match in all directories. For example, "** /foo" matches file or directory "foo" anywhere, the same as pattern "foo".
In your example there is no need to specify path to packages like /data, /domain etc. to every pubspec file. You can just leave "pubspec.lock" or "**/pubspec.lock", this records are equals